src/inference/graph_inference.cpp

```diff
@@ -351,6 +351,56 @@ class GraphNodeAssignment : public Nice2Assignment {
     }
   }
 
+  void GetCandidatesForNode(
+      Nice2Inference* inference, 
+      const int node, 
+      std::vector<std::pair<int, double>>* scored_candidates) {
+    GraphInference* graphInference = static_cast<GraphInference*>(inference);
+    std::vector<int> candidates;
+    GetLabelCandidates(*graphInference, node, &candidates, kMaxPerArcBeamSize);
+
+    scored_candidates->clear();
+    Assignment& nodea = assignments_[node];
+    int original_label = nodea.label;
+    for (size_t i = 0; i < candidates.size() ; i++) {
+      int candidate = candidates[i];
+      nodea.label = candidate;
+      double score = GetNodeScore(*graphInference, node);
+      scored_candidates->push_back(std::pair<int, double>(candidate, score));
+    }
+    nodea.label = original_label;
+
+    std::sort(scored_candidates->begin(), scored_candidates->end(), [](const std::pair<int,double> &left, const std::pair<int,double> &right) {
+      return right.second < left.second;
+    });
+  }
+ 
+  virtual void GetCandidates(
+      Nice2Inference* inference,
+      const int n,
+      Json::Value* response) override {
+    *response = Json::Value(Json::arrayValue);
+    
+    for (size_t i = 0; i < assignments_.size(); ++i) {
+      if (assignments_[i].must_infer) {
+        std::vector<std::pair<int, double>> scored_candidates;
+        GetCandidatesForNode(inference, i, &scored_candidates);
+        Json::Value nodeResults(Json::objectValue);
+        nodeResults["v"] = query_->numberer_.NumberToValue(i);
+        Json::Value nodeCandidates(Json::arrayValue);
+        // Take only the top-n candidates to the response
+        for (size_t j = 0; j < scored_candidates.size() && j < (size_t)((unsigned)n) ; j++) {
+          Json::Value obj(Json::objectValue);
+          obj["label"] = label_set_->GetLabelName(scored_candidates[j].first);
+          obj["score"] = scored_candidates[j].second;
```

Comment:
Just a clarification here: this is the score that comes only from the node and its adjacent edges. It is not the total score. One way to show the total score is to get the total score without updating a label (initial), also the adjacent score of it (old) and then this score (new). The the new total score is initial-old+new. Do you want this or you want the local score only?